src/redux/contact/contact.sagas.js

```diff
@@ -0,0 +1,149 @@
+  updateContact
+} from './contact.operations';
+
+import {
+  setContacts,
+  setContactsLoading,
+  setContact,
+  setContactsError,
+  setContactsCurrentPage,
+  setContactsPagesCount
+} from './contact.actions';
+
+import {
+  GET_CONTACTS,
+  DELETE_CONTACT,
+  ADD_CONTACT,
+  UPDATE_CONTACT,
+  GET_CONTACT
+} from './contact.types';
+
+import {
+  setSnackBarSeverity,
+  setSnackBarStatus,
+  setSnackBarMessage
+} from '../snackbar/snackbar.actions';
+
+import { config } from '../../configs';
+
+const {
+  SUCCESS_ADD_STATUS,
+  SUCCESS_DELETE_STATUS,
+  SUCCESS_UPDATE_STATUS
+} = config.statuses;
+
+const { skip, limit, contactsPerPage } = config.paginationPayload;
+
+function* handleContactsLoad({
+  payload = {
+    skip: 1,
+    limit: 1,
+    contactsPerPage: 1
+  }
+}) {
+  try {
+    yield put(setContactsLoading(true));
+    const contacts = yield call(getContacts, payload.skip, payload.limit);
+    yield put(
+      setContactsPagesCount(Math.ceil(contacts.count / payload.contactsPerPage))
+    );
+    yield put(setContacts(contacts.items));
+    yield put(setContactsLoading(false));
+  } catch (error) {
+    yield call(handleContactsError, error);
+  }
+}
+
+function* handleContactLoad({ payload }) {
+  try {
+    yield put(setContactsLoading(true));
+    const contact = yield call(getContactById, payload);
+    yield put(setContact(contact));
+    yield put(setContactsLoading(false));
+  } catch (error) {
+    yield call(handleContactsError, error);
+  }
+}
+
+function* handleAddContact({ payload }) {
+  try {
+    yield put(setContactsLoading(true));
+    yield call(addContact, payload);
+    const contacts = yield call(getContacts, skip, limit);
+    yield put(
+      setContactsPagesCount(Math.ceil(contacts.count / contactsPerPage))
+    );
+    yield put(setContacts(contacts.items));
+
+    yield put(setSnackBarSeverity('success'));
+    yield put(setSnackBarMessage(SUCCESS_ADD_STATUS));
+    yield put(setSnackBarStatus(true));
+    yield put(push('/'));
+  } catch (error) {
+    yield call(handleContactsError, error);
+  }
+}
+
+function* handleContactDelete({ payload }) {
+  try {
+    yield put(setContactsLoading(true));
+    yield call(deleteContact, payload);
+    const contacts = yield call(getContacts, skip, limit);
+    yield put(
+      setContactsPagesCount(Math.ceil(contacts.count / contactsPerPage))
+    );
+    yield put(setContactsCurrentPage(1));
+    yield put(setContacts(contacts.items));
+
+    yield put(setContactsLoading(false));
+    yield put(setSnackBarSeverity('success'));
+    yield put(setSnackBarMessage(SUCCESS_DELETE_STATUS));
+    yield put(setSnackBarStatus(true));
+  } catch (error) {
+    yield call(handleContactsError, error);
+  }
+}
+
+function* handleContactUpdate({ payload }) {
+  const { id, updatedContact } = payload;
+  try {
+    yield put(setContactsLoading(true));
+    yield call(updateContact, id, updatedContact);
+
+    const contacts = yield call(getContacts, skip, limit);
```

Comment:
Do you really need to fetch all contacts again after updating? I think it is much easier to just update a new updated one in the redux instead of making another request to the backend